### PR TITLE
Build process and development workflow enhancements

### LINKED ISSUE #74: Use Grunt to run a connect server on a watch with live-reload for dev purposes

### LINKED ISSUE #75: Need automated unit test build tasks for the js scripts and automated jshint task run

### PR DESCRIPTION
**Pull request by nelsonomuto**
fixes #74 - configured a server with a watch and live-reload for easier dev workflow
fixes #75 - configured test task to run unit tests

new build and workflow commands:

grunt serve: this will start a server with a watch and livereload for dev purposes.
grunt build: build the code, run the css compilers based on provided target.
grunt jshint: lint the js code.
grunt test: run unit tests on the browser using jasmine framework.

changes to existing source include:
modifications to pass jshint tests:
- double-quotes, long lines, camel case for variable names in ripples.js
- made ripples a global variable in scripts/ripples.js to pass the jshint unused var test.

### CODE REVIEW

**Comment by nelsonomuto** [comment]
This is now current and ensures a working livereload, any modifications to style were simple for testing this and were reverted. Any changes to js source was to ensure adherence to jshintrc. 

Ensure to run an `npm install` so that you have all the build deps before proceeding to test the new commands.

File: test/materialSpec.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+describe('Material', function (){
+
+    //Dummy test just to ensure tests are accurately configured
+    it('jquery should be loaded', function () {
+        expect($).toBeDefined();
+    });
+});
```
Comment (FezVrasta) [inline test/materialSpec.js]: missing new line (maybe we could add some control to check there are newlines at the end of every file)

File: Gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
+module.exports = function(grunt) {
```
Comment (FezVrasta) [inline Gruntfile.js:1]: why have you changed the name from gruntfile to Gruntfile?

Comment (TransitoryBliss) [inline Gruntfile.js:1]: It's considered best practice - grunt checks for both but they suggest having it capitalized.

Comment (FezVrasta) [inline Gruntfile.js:1]: ah oke :+1:

File: test/materialSpec.js (hunk 1 above)
Comment (nelsonomuto) [inline test/materialSpec.js]: that is an old convention to ensure legacy unix tools work consistently. It is not a bug and I advise not spending time implementing any control checks for this.

**Comment by FezVrasta** [comment]
I've merged your PR with the develop branch. Will make some test and push to master when we are ready.

File: test/materialSpec.js (hunk 1 above)
Comment (cncolder) [inline test/materialSpec.js]: what about `.editorconfig`?
this make easy for switch between different project.
need install a [plugin](http://editorconfig.org/#download). support all most editor. include brackets for @FezVrasta .

``` ini
root = true
[*]
insert_final_newline = true
[*.js]
indent_style = space
indent_size = 4
[*.json]
indent_style = space
indent_size = 4
[*.less]
indent_style = space
indent_size = 4
[*.sass]
indent_style = space
indent_size = 4
```

Comment (FezVrasta) [inline test/materialSpec.js]: good idea, brackets has its own .brackets.json but this is cross-editor so it's a much better option. 2faa68a1fa2e7b4c904546413a05ba6b985e6505

**Comment by TransitoryBliss** [comment]
:+1:

**Comment by TransitoryBliss** [comment]
Is there a reason why we don't use a headless browser for the tests? I

**Comment by FezVrasta** [comment]
I've never used phantomjs or similar to run tests so I've not experience :\

**Pull request merged**